src/braket/ahs/atom_arrangement.py

```diff
@@ -0,0 +1,120 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from decimal import Decimal
+from enum import Enum
+from numbers import Number
+from typing import Iterator, List, Tuple, Union
+
+import numpy as np
+
+from braket.ahs.discretization_types import DiscretizationError, DiscretizationProperties
+
+
+class SiteType(Enum):
+    VACANT = "Vacant"
+    FILLED = "Filled"
+
+
+@dataclass
+class AtomArrangementItem:
+    coordinate: Tuple[Number, Number]
+    site_type: SiteType
+
+    def _validate_coordinate(self):
+        if len(self.coordinate) != 2:
+            raise ValueError(f"{self.coordinate} must be of length 2")
+        for idx, num in enumerate(self.coordinate):
+            if not isinstance(num, Number):
+                raise TypeError(f"{num} at position {idx} must be a number")
+
+    def _validate_site_type(self):
+        allowed_site_types = {SiteType.FILLED, SiteType.VACANT}
+        if self.site_type not in allowed_site_types:
+            raise ValueError(f"{self.site_type} must be one of {allowed_site_types}")
+
+    def __post_init__(self):
+        self._validate_coordinate()
+        self._validate_site_type()
+
+
+class AtomArrangement:
+    def __init__(self):
+        """Represents a set of coordinates that can be used as a register to an
+        AnalogHamiltonianSimulation.
+        """
+        self._sites = []
+
+    def add(
+        self, coord: Union[Tuple[Number, Number], np.ndarray], site_type: SiteType = SiteType.FILLED
```

Comment:
I'm confused why we have different naming conventions across methods. There is `coord` here but `coordinate_index` in the `coordinate_list` method. We should use a uniform naming convention (either `coordinate`/`coordinate_index` or `coord`/`coord_index`) across our entire API.